packages/cli-tool/src/commands/generate/index.ts

```diff
@@ -27,39 +29,29 @@ export default class Generate extends Command {
   ];
 
   public async run(): Promise<void> {
-    const { args } = await this.parse(Generate);
-    const appName = args.appName;
-    const versionControlChoices = getChoices(VERSION_CONTROL_OPTIONS);
-    const questions = [
-      {
-        type: 'list',
-        name: 'versionControl',
-        message: 'Select a version control service:',
-        choices: versionControlChoices,
-      },
-    ];
+    const {
+      args: { appName, template },
+    } = await this.parse(Generate);
+
     const answers = await Inquirer.prompt(questions);
 
     try {
       this.log(
         `Generating Nimble React app with the project name: ${appName}!`,
       );
-      const result = await this.config.runHook('initialize', {
-        appName,
-        template: args.template,
-      });
-
-      if (hookFailed(result)) {
-        cli.info(
-          'Something went wrong while generating the cra-template...',
-          formatHookErrorMsg(result),
-        );
-        return;
-      }
 
+      await initializeCraApp(appName, template);
       setVersionControl(appName, answers.versionControl);
+      await setUIFramework(appName, answers.uiFramework);
+
+      // Clean files after all steps
+      this.cleanFiles(appName);
     } catch (error) {
       this.error(error as string | Error);
     }
   }
+
+  cleanFiles = (appName: string): void => {
+    fs.rmSync(`${appName}/.add-ons`, { recursive: true });
```

Comment:
Sync = it is a Synchronous function (the opposite of **A**sync ;-) ). So even if we wanted, we could not use await with this function.

Reply:
Updated to use the async version `fs.rm` so that we can prevent the thread to be blocked in case there is an issue with the file deletion, as you recommended ~

Thanks!!! 🙌